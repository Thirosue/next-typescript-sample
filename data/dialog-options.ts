import React from 'react'

export type DialogOptions = {
  html?: boolean
  alert?: boolean
  icon?: 'info' | 'warn' | 'alert'
  title: string
  description: React.ReactNode
  confirmationText?: string
  cancellationText?: string
}
