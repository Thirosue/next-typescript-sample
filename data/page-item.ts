export type PageItem = {
  page: number
  totalPage?: number
  totalCount: number
  perPage: number
}
