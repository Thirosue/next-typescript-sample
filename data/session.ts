export type Session = {
  username?: string
  sub: string
  jwtToken: string
  email_verified?: boolean
}
