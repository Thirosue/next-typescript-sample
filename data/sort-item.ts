export type SortItem = {
  key: string
  order: '' | 'asc' | 'desc'
}
