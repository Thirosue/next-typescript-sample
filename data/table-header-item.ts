export type TableHeaderItem = {
  key?: string
  label: string
  sortable?: boolean
  onClick?: () => Promise<void>
}
