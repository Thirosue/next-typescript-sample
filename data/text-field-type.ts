export enum TextFieldType {
  Text = 'text',
  Number = 'number',
  Email = 'email',
  Password = 'password',
  Tel = 'tel',
  Url = 'url',
}
