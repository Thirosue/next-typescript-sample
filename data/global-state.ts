import { Session } from './session'

export type GlobalState = {
  session: Session
}
