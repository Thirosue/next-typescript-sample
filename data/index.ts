import { TextFieldType } from './text-field-type'

export { TextFieldType }
