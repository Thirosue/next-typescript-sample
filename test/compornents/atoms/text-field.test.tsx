import React from 'react'
import { render } from '../../testUtils'
import TextField from '../../../components/atoms/text-field'
import { TextFieldType } from '../../../data'

describe('TextField components', () => {
  it('matches snapshot', () => {
    const { asFragment } = render(
      <TextField
        label={'test'}
        type={TextFieldType.Email}
        value={'defalut'}
        classes={['test1', 'test2']}
        error={false}
        onChange={() => console.log('hoge')}
      />,
      {}
    )
    expect(asFragment()).toMatchSnapshot()
  })

  // it('clicking button triggers alert', () => {
  //     const { getByText } = render(<Home />, {})
  //     window.alert = jest.fn()
  //     fireEvent.click(getByText('Test Button'))
  //     expect(window.alert).toHaveBeenCalledWith('With typescript and Jest')
  // })
})
