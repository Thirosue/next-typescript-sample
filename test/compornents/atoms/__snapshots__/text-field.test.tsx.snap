// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TextField components matches snapshot 1`] = `
<DocumentFragment>
  <label
    class="block test1 test2"
  >
    <span
      class="text-gray-700 text-sm"
    >
      test
    </span>
    <input
      class="mt-1 border-gray-300 block w-full rounded-md focus:border-indigo-600"
      type="email"
      value="defalut"
    />
  </label>
</DocumentFragment>
`;
